.circleci/initjob/30-set-collector-version.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+shared_env=$1
+dockerized_cache_tag=$2
+build_tag=$3
+TAG=$4
+BRANCH=$5
+SOURCE_ROOT=$6
+
+check_use_collector_builder_cache() {
+    use_cache="true"
+
+    if [[ -n "$TAG" ]]; then
+         use_cache="false"
+    fi
+
+    if [[ "$BRANCH" == "master" ]]; then
+         use_cache="false"
+    fi
+
+    if [[ -f pr-metadata/labels/build-builder-image ]]; then
+         use_cache="false"
+    fi
+
+    if [[ -f pr-metadata/labels/valgrind-unit-tests ]]; then
+         use_cache="false"
+    fi
+
+    if [[ -f pr-metadata/labels/valgrind-integration-tests ]]; then
+         use_cache="false"
+    fi
+
+    echo "${use_cache}"
+}
+
+COLLECTOR_VERSION="$(make -s -C "${SOURCE_ROOT}" tag)"
+COLLECTOR_TAG="${COLLECTOR_VERSION}"
+GCLOUD_SSH_FINGERPRINT="$(echo "${GCLOUD_SSH_KEY_PUB}" | awk '{print $2}' | base64 -d | md5sum | awk '{print $1}')"
+GCP_SSH_KEY_FILE="${HOME}/.ssh/id_rsa_${GCLOUD_SSH_FINGERPRINT}"
+
+cat >> "$shared_env" <<- EOF
+    export COLLECTOR_VERSION="${COLLECTOR_VERSION}"
+    export COLLECTOR_TAG="${COLLECTOR_TAG}"
+    export GCP_SSH_KEY_FILE="${GCP_SSH_KEY_FILE}"
+EOF
+
+if [[ -z "$TAG" && "$BRANCH" != "master" ]]; then
+    echo "export COLLECTOR_APPEND_CID=true" >> "$shared_env"
+fi
+
+if [[ "$(check_use_collector_builder_cache)" == "true" ]]; then
+    echo "export COLLECTOR_BUILDER_TAG=cache" >> "$shared_env"
+else
+    cat >> "$shared_env" <<- EOF
+    export COLLECTOR_BUILDER_TAG="${build_tag}"
+    export BUILD_BUILDER_IMAGE=true
+EOF
+fi
+
+COLLECTOR_DRIVERS_CACHE="$dockerized_cache_tag"
+COLLECTOR_DRIVERS_TAG="${build_tag}"
```

Comment:
Any particular reason why one value is surrounded by curly braces, and another isn't?

Reply:
No. There is a lot of inconsistency here. Most places that have curly braces don't need them. It is only really needed if there is some text after the variable name that would be confused for a variable name. E.g $var1_$var2 vs ${var1}_${var2}. The CircleCI config already had this problem of having curly braces where it was not needed and I never really decided what to do about it, though these PRs would have been a good opportunity to come up with a guide for this and make everything consistent. This can be fixed in a future PR.